src/dom/Value.ts

```diff
@@ -217,10 +217,21 @@ export interface Value {
   ): boolean;
 
   /**
-   * For checking non-strict equivalence of an Ion Values represented by dom.Value with the other value represented
-   * by an JS object.
+   * Compares this instance of dom.Value to the provided value and returns true
+   * if they are equal. If `other` is a dom.Value, this comparison checks for structural
+   * (or "non-strict") equivalence. If `other` is a native JS value, this comparison will
+   * convert it to an Ion value and then check for structural equivalence.
```

Comment:
> If `other` is a native JS value, this comparison will convert it to an Ion value and then check for structural equivalence.

I know this was originally my phrasing suggestion, but I think we determined that it's not accurate. We do the comparisons without converting the JS value to Ion first.